# Experiment-16
## Theory
Exceptions are runtime errors or unusual conditions a program faces during execution. In C++, exception handling helps manage these situations, allowing the program to recover or exit smoothly.
C++ uses these keywords:
- try : Defines a block of code that may throw an exception.
- catch: Handles exceptions thrown by the try block.
- throw: Triggers an exception.
When an exception occurs, control moves to the throw statement, which directs it to the appropriate catch block to handle the error, allowing the program to either recover or terminate.
## Program 1
## Aim
Prompt the user to enter a positive number. If the user enters a positive number, display that number. If the user enters a negative number throw an exception.

## Software used
Visual Studio Code

## Output
![image](https://github.com/user-attachments/assets/20d5cfac-9770-4ab5-b198-09009f7d884f)

![image](https://github.com/user-attachments/assets/18e27fed-b6db-47ba-a98c-c07c84912900)

## Conclusion
We applied exception handling in C++ to check whether a negative or positive integer.

## Program 2
## Aim
Prompt the user to enter a year greater than 2000. If the user enters the year greater than 2000, then display that year. If the user enters a year less than 2000 throw an exception.

## Software used
Visual Studio Code

## Output
![image](https://github.com/user-attachments/assets/04c1588f-bddd-4bbd-9ee2-7ef30bc7bf44)

![image](https://github.com/user-attachments/assets/a97e3c97-fd87-49cf-b169-c99702660bcd)

## Conclusion
We applied exception handling in C++ to check whether the entered year belongs to the 21st century or not.
